axolotl-web/src/store/store.js

```diff
@@ -283,55 +306,83 @@ export default createStore({
-          this.commit("SET_CONFIG_GUI", messageData["Gui"]);
-        }
-        else if (Object.keys(messageData)[0] === "DarkMode") {
-          this.commit("SET_CONFIG_DARK_MODE", messageData["DarkMode"]);
-        }
-        else if (Object.keys(messageData)[0] === "CurrentChat") {
-          this.commit("SET_CURRENT_CHAT", messageData["CurrentChat"]);
-        }
-        else if (Object.keys(messageData)[0] === "Type") {
-          this.commit("SET_REQUEST", messageData);
-        }
-        else if (Object.keys(messageData)[0] === "FingerprintNumbers") {
-          this.commit("SET_FINGERPRINT", messageData);
-        }
-        else if (Object.keys(messageData)[0] === "Error") {
-          this.commit("SET_ERROR", messageData.Errorx);
-        }
-        else if (Object.keys(messageData)[0] === "OpenChat") {
-          this.commit("OPEN_CHAT", messageData["OpenChat"]);
-        }
-        else if (Object.keys(messageData)[0] === "UpdateCurrentChat") {
-          this.commit("UPDATE_CURRENT_CHAT", messageData["UpdateCurrentChat"]);
-        }
-        else {
+          this.commit("SET_ERROR", messageData.Error);
+        }
+        switch (Object.keys(messageData)[0]) {
+          case "ChatList":
+            const chats = messageData.ChatList
+            if (messageData.LastMessages !== undefined) {
+              const lastMessages = {};
+              for (let i = 0; i < messageData.LastMessages.length; i++) {
+                lastMessages[messageData.LastMessages[i].SID] = messageData.LastMessages[i];
+              }
+              chats.sort((a, b) => {
+                if (typeof lastMessages[a.ID] === "undefined" || typeof lastMessages[b.ID] === "undefined") {
+                  return 0;
+                }
+                if (lastMessages[a.ID].SentAt < lastMessages[b.ID].SentAt) {
+                  return 1;
+                }
+                if (lastMessages[a.ID].SentAt > lastMessages[b.ID].SentAt) {
+                  return -1;
+                }
+                return 0;
+              }
+              );
+              this.commit("SET_CHATLIST", chats);
+              this.commit("SET_LASTMESSAGES", lastMessages);
+            } else {
+              this.commit("SET_CHATLIST", chats);
+              this.commit("SET_LASTMESSAGES", messageData.LastMessages);
+            }
+            this.commit("SET_SESSIONNAMES", messageData.SessionNames);
+            break;
+          case "MessageList":
+            this.commit("SET_MESSAGELIST", messageData.MessageList);
+            break;
+          case "ContactList":
+            this.commit("SET_CONTACTS", messageData.ContactList);
+            break;
+          case "MoreMessageList":
+            this.commit("SET_MORE_MESSAGELIST", messageData.MoreMessageList);
+            break;
+          case "DeviceList":
+            this.commit("SET_DEVICELIST", messageData.DeviceList);
+            break;
+          case "MessageRecieved":
```

Comment:
Typo: MessageRecieved --> MessageReceived
also in the next line and maybe regarding the function MessageRecieved (-->Received)